plugins/carbon_black_defense/komand_carbon_black_defense/actions/get_details_for_specific_event/action.py

```diff
@@ -18,42 +15,35 @@ def __init__(self):
             output=GetDetailsForSpecificEventOutput(),
         )
 
-    def run(self, params={}):
-        host = self.connection.host
-        token = self.connection.token
-        connector = self.connection.connector
+    def run(self, params=None):
+
+        if params is None:
+            params = {}
         event_id = params.get(Input.EVENT_ID)
+        id_ = self.connection.get_job_id_for_detail_search(event_id=event_id)
+        self.logger.info(f"The id is: {id_}")
+        if id_ is None:
+            return {Output.SUCCESS: False, Output.EVENTINFO: {}}
+        detail_search_status = self.connection.check_status_of_detail_search(id_)
+
+        # check if status of
+        # detail search is complete by checking if the completed property
+        # in results is not equal to the contacted property
+        for _ in range(0, 9999):
+            if not detail_search_status:
+                detail_search_status = self.connection.check_status_of_detail_search(id_)
+                t1 = datetime.now()
+                if (datetime.now() - t1).seconds > 5:
+                    break
+                time.sleep(2)
+            else:
+                break
+
+            response = self.connection.retrieve_results_for_detail_search(job_id=id_)
+            data = insightconnect_plugin_runtime.helper.clean(response)
+            self.logger.info(f"The data is: {data}")
```

Comment:
Also want to remove this overly verbose debug comment

Reply:
removed debug comment